### DISCUSSION TITLE
Some questions

### DISCUSSION THREAD

**Discussion by rieje** [romkatv/powerlevel10k#2757] [General]
* I use `GIT_DIR="$HOME"/.dotfiles.git/ GIT_WORK_TREE="$HOME"` to manage dotfiles and toggle this dotfile environment on with ` export GIT_DIR="$HOME"/.dotfiles.git/ GIT_WORK_TREE="$HOME"`. `vcs` segment does not detect this (not that I expect to). Is it possible to still use this segment somehow or should I try to make a custom segment for it? In the event that I'm in a git repo while the dotfile environment is on , `vcs` info should be prioritized showing the repo I'm on over vcs info for the dotfile environment.

* If I want the prompt to use colors from my terminal theme, I have to hardcode this (by looking at the theme's defined colors) and there's not a way to have a prompt to pick from the set of colors from the theme to automatically use, right?

* Can the location of cache and .p10k.zsh be changed? I guess .p10k.zsh unless one decides to run the wizard again can but not sure about the cache, which I would like to dump to `$XDG_CACHE_HOME/zsh`.

Much appreciated.

**Comment by romkatv** [comment]
For the `vcs` question, first see https://github.com/romkatv/powerlevel10k/blob/master/README.md#why-is-git-status-from-homegit-not-displayed-in-prompt. If removing `POWERLEVEL9K_VCS_DISABLED_WORKDIR_PATTERN` does not help, flip `core.bare` of the repo to `false` and put `core.worktree` in the config. Something like this:

```zsh
GIT_DIR=~/.dotfiles.git git config core.bare false
GIT_DIR=~/.dotfiles.git git config core.worktree ~
```

For colors, see https://github.com/romkatv/powerlevel10k/blob/master/README.md#how-do-i-change-prompt-colors.

The location of the cache cannot be changed.

The location of `.p10k.zsh` can be changed. Move it anywhere you like and update the `source` line in `.zshrc`. You can even run `p10k configure` afterwards and it'll override your config (after asking for confirmation).

**Reply by rieje** [reply]
Thanks. `GIT_DIR=~/.dotfiles.git git config core.bare false` alone worked though I'm not sure if there are any implications on the repo or why it works.

**Reply by romkatv** [reply]
There are no negative implications. I'm guessing you've copied the core.bare bit from someone else. I've seen lots of dotfiles repos that have this bit set for no reason.

**Reply by rieje** [reply]
It's set from using it as a bare repo with `git init --bare`. Dotfiles are managed using:

 * export/unset `GIT_DIR=~/.dotfiles.git GIT_WORK_TREE="$HOME"` to toggle environment for git commands
 * `showUntrackedFiles = no`: only show explicitly tracked files

**Reply by romkatv** [reply]
Yep, it really doesn't need to be a bare repo.

**Reply by rieje** [reply]
Quick question:

I would like an icon indiator when I'm working in the dotfile environment (`GIT_DIR`/`WORK_TREE` set). Is this condition appropriate to integrate with `my_git_formatter` function? I added `[[ -d "$GIT_DIR" ]] && res+=" "` to the end of the function but it has no effect. Or maybe my dotfiles function can affect the prompt for the same effect. The lazy way would be a custom segment to check for the values of these variables on every command but I might want to expand on this a little more in the future so prefer the "correct" way.

**Reply by romkatv** [reply]
Your change to `my_git_formatter` checks for `GIT_DIR` within zsh, where it's always unset.

Here's how I do it: https://github.com/romkatv/zsh4humans/blob/master/tips.md#managing-dotfiles.

**Reply by Syphdias** [reply]
> I've seen lots of dotfiles repos that have this bit set for no reason.

Guilty
